src/node.cc

```diff
@@ -4322,6 +4401,26 @@ static void StartNodeInstance(void* arg) {
       } while (more == true);
     }
 
+    Local<Value> promise_keys_set =
+        env->promise_unhandled_reject_keys().As<Value>();
+    Local<Function> convert = env->array_from();
+    Local<Value> ret = convert->Call(env->context(),
+        Null(env->isolate()), 1, &promise_keys_set).ToLocalChecked();
+    Local<Array> promise_keys = ret.As<Array>();
+    uint32_t key_count = promise_keys->Length();
+    Local<NativeWeakMap> unhandled_reject_map =
+        env->promise_unhandled_reject_map();
+
+    for (uint32_t key_iter = 0; key_iter < key_count; key_iter++) {
+      Local<Value> key = promise_keys->Get(env->context(),
+                                           key_iter).ToLocalChecked();
+
+      if (unhandled_reject_map->Has(key)) {
+        ReportPromiseRejection(isolate, unhandled_reject_map->Get(key));
+        exit(1);
+      }
+    }
```

Comment:
>  Is it possible to filter at this point and only crash on unreferenced rejections?

No. VMs do not reference count for performance reasons. If they did, all of this would use it.

(And good lord I wish they did for this.)

--

This case is, imo, an acceptable limitation.

If someone is advanced enough to know what they are doing like that, I'm betting they can add their own `uncaughtRejection` handler, rather than relying on people who are new to the platform to know about some error handler before they can even see mistakes they may have made.